python/pyspark/find_spark_home.py

```diff
@@ -59,15 +59,15 @@ def is_spark_home(path):
             paths.append(os.path.join(module_home, "../../"))
         except ImportError:
             # Not pip installed no worries
-            True
+            pass
 
     # Normalize the paths
     paths = [os.path.abspath(p) for p in paths]
 
     try:
         return next(path for path in paths if is_spark_home(path))
     except StopIteration:
-        print("Could not find valid SPARK_HOME while searching %s".format(paths), file=sys.stderr)
+        print("Could not find valid SPARK_HOME while searching {0}".format(paths), file=sys.stderr)
```

Comment:
Added, sorry (got left out when I was working on the change to for edit mode tests and I reset part of the way through).